### PR TITLE
Move S3Client creation from S3StorageConfig

### PR DESCRIPTION
**Pull request by ivanyu**
Creating the client shouldn't really be done in the config class. This commit extracts it into a separate class.

### CODE REVIEW

File: storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilder.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.core.internal.http.loader.DefaultSdkHttpClientBuilder;
+import software.amazon.awssdk.http.SdkHttpConfigurationOption;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.utils.AttributeMap;
+
+class S3ClientBuilder {
+    static S3Client build(final S3StorageConfig config) {
```
Comment (ivanyu) [inline storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilder.java:29]: This method was extracted from `S3StorageConfig` with minimal changes necessary

File: storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    void minimalConfig() {
+        final var configs = Map.of(
+            "s3.bucket.name", "b",
+            "s3.region", TEST_REGION.id()
+        );
+        final var config = new S3StorageConfig(configs);
+        final var s3Client = S3ClientBuilder.build(config);
+        final var clientConfiguration = s3Client.serviceClientConfiguration();
+        assertThat(clientConfiguration.region()).isEqualTo(TEST_REGION);
+        assertThat(clientConfiguration.endpointOverride().orElse(null)).isNull();
```
Comment (AnatolyPopov) [inline storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java:43]:
```suggestion
        assertThat(clientConfiguration.endpointOverride()).isNotPresent();
```

File: storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            "aws.access.key.id", username,
+            "aws.secret.access.key", password,
+            "aws.certificate.check.enabled", "false",
+            "aws.checksum.check.enabled", "true");
+
+        final var config = new S3StorageConfig(configs);
+        final var s3Client = S3ClientBuilder.build(config);
+        final var clientConfiguration = s3Client.serviceClientConfiguration();
+        assertThat(clientConfiguration.region()).isEqualTo(TEST_REGION);
+        assertThat(clientConfiguration.endpointOverride().orElse(null)).isEqualTo(URI.create("http://minio"));
```
Comment (AnatolyPopov) [inline storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java:110]:
```suggestion
        assertThat(clientConfiguration.endpointOverride()).hasValue(URI.create("http://minio"));
```

File: storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+            "s3.endpoint.url", MINIO_URL,
+            "s3.path.style.access.enabled", false,
+            "s3.multipart.upload.part.size", partSize,
+            "aws.credentials.provider.class", customCredentialsProvider.getName());
+
+        final var config = new S3StorageConfig(configs);
+        final var s3Client = S3ClientBuilder.build(config);
+        final var clientConfiguration = s3Client.serviceClientConfiguration();
+        assertThat(clientConfiguration.region()).isEqualTo(TEST_REGION);
+        assertThat(clientConfiguration.endpointOverride().orElse(null)).isEqualTo(URI.create("http://minio"));
```
Comment (AnatolyPopov) [inline storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java:85]:
```suggestion
        assertThat(clientConfiguration.endpointOverride()).hasValue(URI.create("http://minio"));
```

File: storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+            "s3.bucket.name", "b",
+            "s3.region", TEST_REGION.id(),
+            "s3.endpoint.url", MINIO_URL,
+            "s3.path.style.access.enabled", true
+        );
+        final var config = new S3StorageConfig(configs);
+        final var s3Client = S3ClientBuilder.build(config);
+        final var clientConfiguration = s3Client.serviceClientConfiguration();
+        assertThat(clientConfiguration.region()).isEqualTo(TEST_REGION);
+        assertThat(clientConfiguration.endpointOverride().orElse(null)).isEqualTo(URI.create("http://minio"));
```
Comment (AnatolyPopov) [inline storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java:62]:
```suggestion
        assertThat(clientConfiguration.endpointOverride()).hasValue(URI.create("http://minio"));
```

File: storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java (hunk 2 above)
Comment (ivanyu) [reply storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java]: Fixed

File: storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+            "s3.bucket.name", "b",
+            "s3.region", TEST_REGION.id(),
+            "s3.api.call.timeout", 5000,
+            "s3.api.call.attempt.timeout", 1000
+        );
+        final var config = new S3StorageConfig(configs);
+        final var s3Client = S3ClientBuilder.build(config);
+        final var clientConfiguration = s3Client.serviceClientConfiguration();
+        assertThat(clientConfiguration.region()).isEqualTo(TEST_REGION);
+        assertThat(clientConfiguration.endpointOverride().orElse(null)).isNull();
```
[the remark this answers is not in the document]
Comment (ivanyu) [reply storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java]: Fixed

File: storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java (hunk 3 above)
Comment (ivanyu) [reply storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java]: Fixed

File: storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java (hunk 4 above)
Comment (ivanyu) [reply storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java]: Fixed

File: storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java (hunk 5 above)
Comment (ivanyu) [reply storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java]: Fixed

**Review by jeqo — APPROVED** [review]
LGTM

**Review by AnatolyPopov — APPROVED** [review]
LGTM

**Pull request merged**